### PR TITLE
Manage legend patch shapes through style manager

### PR DESCRIPTION
**Pull request by nyalldawson**
This PR finalises the recent legend patch shape customisation work, by adding in support for custom legend patch shapes to style manager!

Now we just need a really nice set of default patch shapes to ship!!

Peek 2020-04-22 10-18

### CODE REVIEW

File: src/core/symbology/qgsstyle.cpp [hunk 1]
```diff
@@ -868,25 +902,7 @@ bool QgsStyle::saveLabelSettings( const QString &name, const QgsPalLayerSettings
...
-  }
-
-  mCachedTags[ LabelSettingsEntity ].remove( name );
-  mCachedFavorites[ LabelSettingsEntity ].remove( name );
-
-  emit labelSettingsRemoved( name );
-  emit entityRemoved( LabelSettingsEntity, name );
-
-  return true;
+  return removeEntityByName( LabelSettingsEntity, name );
```
Comment (nirvn) [inline src/core/symbology/qgsstyle.cpp:905]: Nice code de-duplication.

File: src/gui/qgslegendpatchshapebutton.cpp [hunk 2]
```diff
@@ -187,6 +188,35 @@ void QgsLegendPatchShapeButton::prepareMenu()
   mMenu->addAction( defaultAction );
   connect( defaultAction, &QAction::triggered, this, [ = ] { setToDefault(); emit changed(); } );
 
+  mMenu->addSeparator();
+
+  QStringList patchNames = QgsStyle::defaultStyle()->symbolsOfFavorite( QgsStyle::LegendPatchShapeEntity );
```
Comment (nirvn) [inline src/gui/qgslegendpatchshapebutton.cpp:193]: Nice use of favorites. I've been meaning to do that for our symbol buttons too.

**Review by nirvn — APPROVED** [review]

File: src/gui/qgslegendpatchshapebutton.cpp (hunk 2 above)
Comment (nyalldawson) [reply src/gui/qgslegendpatchshapebutton.cpp:193]: I copied it from your color ramp button work :stuck_out_tongue:

Comment (nirvn) [reply src/gui/qgslegendpatchshapebutton.cpp:193]: :bow:

**Comment by isaomatsunami** [comment]
Nice improvement! Thank you. But It's so hard to translate this "Patch". I know Patch is Patch.😁

**Pull request merged**